isthmus/src/main/java/io/substrait/isthmus/SubstraitRelNodeConverter.java

```diff
@@ -89,22 +104,133 @@ public RelNode visit(NamedScan namedScan) throws RuntimeException {
+    var operator =
+        aggregateFunctionConverter.getSqlOperatorFromSubstraitFunc(
+            measure.getFunction().declaration().key(), measure.getFunction().outputType());
+    if (!operator.isPresent()) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "Unable to find binding for call %s", measure.getFunction().declaration().name()));
+    }
+    List<Integer> argIndex = new ArrayList<>();
+    for (RexNode arg : arguments) {
+      if (!(arg instanceof RexInputRef)) {
+        throw new UnsupportedOperationException(
+            String.format(
+                "aggregation function only support column as argument. % is not support", arg));
+      }
+      argIndex.add(((RexInputRef) arg).getIndex());
+    }
+
+    boolean distinct =
+        measure.getFunction().invocation()
+            == AggregateFunction.AggregationInvocation.AGGREGATION_INVOCATION_DISTINCT;
+
+    SqlAggFunction aggFunction;
+    RelDataType returnType =
+        TypeConverter.convert(relOptCluster.getTypeFactory(), measure.getFunction().getType());
+
+    boolean approximate = false;
+    if (operator.get() == SqlStdOperatorTable.COUNT) {
+      aggFunction = SqlStdOperatorTable.COUNT;
+    } else if (operator.get() == SqlStdOperatorTable.SUM) {
+      aggFunction = SqlStdOperatorTable.SUM;
+    } else if (operator.get() == SqlStdOperatorTable.AVG) {
+      aggFunction = SqlStdOperatorTable.AVG;
+    } else if (operator.get() == SQL_OPERATOR_APPROX_COUNT_DISTINCT) {
+      aggFunction = SqlStdOperatorTable.COUNT;
+      approximate = true;
+      distinct = true;
+    } else {
+      throw new UnsupportedOperationException(
+          String.format(
+              "Unsupported agg operator % for agg function % ",
+              operator.get(), measure.getFunction().declaration().name()));
+    }
+
+    int filterArg = -1;
+    if (measure.getPreMeasureFilter().isPresent()) {
+      RexNode filter = measure.getPreMeasureFilter().get().accept(expressionRexConverter);
+      if (!(filter instanceof RexInputRef)) {
+        // Calcite's AggregateCall only allow agg filter to be a direct filter from input
+        throw new UnsupportedOperationException(
+            String.format(
+                "Unsupported filter expression % for agg function % ",
+                filter, measure.getFunction().declaration().name()));
+      }
+      filterArg = ((RexInputRef) filter).getIndex();
+    }
+
+    return AggregateCall.create(
+        aggFunction,
+        distinct,
+        approximate,
+        false,
+        argIndex,
+        filterArg,
+        null,
+        RelCollations.EMPTY,
+        returnType,
+        null);
   }
 
   @Override
   public RelNode visitFallback(Rel rel) throws RuntimeException {
     throw new UnsupportedOperationException(
         String.format(
-            "Rel $ of type %s not handled by visitor type %s.",
+            "Rel % of type %s not handled by visitor type %s.",
```

Comment:
What does "%" do as opposed to "%s" in format string? Just curious

Reply:
Thanks for catching this bug.  My eyes were not sharp enough to recognize these typos.  Changed to `%s` in all error messages.